packages/@aws-cdk/aws-batch/lib/task-definition.ts

```diff
@@ -0,0 +1,20 @@
+import * as ecs from '@aws-cdk/aws-ecs';
+import * as iam from '@aws-cdk/aws-iam';
+
+export interface TaskDefinitionProps {
+  readonly executionRole: iam.IRole;
+}
+
+export class TaskDefinition {
+  public readonly executionRole: iam.IRole;
+  constructor(props: TaskDefinitionProps) {
+    this.executionRole = props.executionRole;
+  }
+
+  // tslint:disable-next-line: no-empty
+  public _linkContainer(_container: ecs.ContainerDefinition) {}
+
+  public obtainExecutionRole(): iam.IRole {
+    return this.executionRole;
+  }
+}
```

Comment:
Could you move this to the same file that `JobDefinitionImageConfig` is? They're very tightly coupled.

Reply:
Done 👍